### DISCUSSION TITLE
Is there a useLocation() equivalent?

### DISCUSSION THREAD

**Discussion by manu-remsense** [TanStack/router#1008] [General]
Hey guys I've been trying to find an equivalent to react-router's useLocation, however I've found no such function exists in the current version. I've seen few posts from a year+ ago using this hook, but I assume it was removed now.

I was planning on using useLocation() in order to construct breadcrumbs component but haven't found a good way of doing it.

I have seen the breadcrumbs section where they are using react router context, but it seemed a little complicated compared to an old method of using useLocation and getting path segments,

Any help or advice would be appreciated, thanks 🙏 !

**Comment by hugo-cardoso** [comment]
You can use `useRouterState` hook and access location property.

**Reply by manu-remsense** [reply]
Alright thanks, this is what I ended up doing based on another discussion post, but wasn't sure if there was a better way of doing it, cheers
